### PR TITLE
Add AdditionalData node to the AuthoriseResponse

### PR DESCRIPTION
**Pull request by lbailly**
Add Alias / AliasType fields to AdditionalData to read values returned by an authorisation request

### CODE REVIEW

File: recurring.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	ShopperReference string `json:"shopperReference,omitempty"`
+	// Not mandatory
+	Recurring *Recurring `json:"recurring,omitempty"`
+}
+
+// RecurringDetailsResult structure to hold the RecurringDetails
+type RecurringDetailsResult struct {
+	CreationDate string `json:"creationDate"`
+	Details      []struct {
+		RecurringDetail struct {
```
Comment (zhutik) [inline recurring.go]: Maybe it's better to decouple RecurringDetail?
Card structure already exists, so maybe it could be used there as well?

**Comment by zhutik** [comment]
@lbailly thanks a lot for PR! Super useful!

I'm also created a playgroud for this library https://github.com/zhutik/adyen-api-go-example maybe you can update it as well after this one get merged? It's useful for testing and verification, so make everything works as expected.

File: recurring.go (hunk 1 above)
Comment (lbailly) [reply recurring.go]: I'll decouple RecurringDetail.
For Card, the existing struct uses int for expiryMonth and expiryYear but JSon contains string values. So the unmarshalling process throws an errror : "json: cannot unmarshal string into Go struct field Card.expiryMonth of type int"

Below an example of Json returned by Adyen :
{"creationDate":"2017-09-27T16:30:23+02:00","details":[{"RecurringDetail":{"acquirer":"TestPmmAcquirer","acquirerAccount":"TestPmmAcquirerAccount","additionalData":{"cardBin":"411111"},"alias":"H167852639363479","aliasType":"Default","card":{"expiryMonth":"8","expiryYear":"2018","holderName":"test","number":"1111"},"contractTypes":["RECURRING"],"creationDate":"2017-09-27T16:30:23+02:00","firstPspReference":"8815065226234138","paymentMethodVariant":"visa","recurringDetailReference":"8315065226234554","variant":"visa"}},{"RecurringDetail":{"acquirer":"TestPmmAcquirer","acquirerAccount":"TestPmmAcquirerAccount","additionalData":{"cardBin":"497794"},"alias":"F314134953496447","aliasType":"Default","card":{"expiryMonth":"8","expiryYear":"2018","holderName":"test","number":"9497"},"contractTypes":["RECURRING"],"creationDate":"2017-09-27T16:58:08+02:00","firstPspReference":"8815065242885772","paymentMethodVariant":"visagold","recurringDetailReference":"8415065242885172","variant":"visa"}},{"RecurringDetail":{"acquirer":"TestPmmAcquirer","acquirerAccount":"TestPmmAcquirerAccount","additionalData":{"cardBin":"558555"},"alias":"A829629780526225","aliasType":"Default","card":{"expiryMonth":"8","expiryYear":"2018","holderName":"test","number":"5583"},"contractTypes":["RECURRING"],"creationDate":"2017-09-28T11:07:09+02:00","firstPspReference":"8515065896290058","paymentMethodVariant":"mc","recurringDetailReference":"8315065896294334","variant":"mc"}}],"shopperReference":"346","invalidOneclickContracts":"false"}

Comment (zhutik) [reply recurring.go]: than I guess it's better to update card structure to string type for month and year
https://docs.adyen.com/developers/api-reference/payments-api#card 
f.e 1 (January) should be 01 as shown on API reference

**Comment by zhutik** [comment]
@lbailly 

```
recurring.go:21:6: exported type RecurringDetail should have comment or be unexported
recurring_gateway.go:11:1: comment on exported method RecurringGateway.ListRecurringDetails should be of the form "ListRecurringDetails ..."
```

2 lint errors to fix.

I've fixed unit tests on TravisCI for pull requests from forks, so now it's more useful

File: adyen.go [hunk 2]
```diff
@@ -97,13 +103,13 @@ func (a *Adyen) SetCurrency(currency string) {
...
 // execute request on Adyen side, transforms "requestEntity" into JSON representation
-func (a *Adyen) execute(method string, requestEntity interface{}) (*Response, error) {
+func (a *Adyen) execute(method string, service string, requestEntity interface{}) (*Response, error) {
 	body, err := json.Marshal(requestEntity)
 	if err != nil {
 		return nil, err
 	}
 
-	url := a.adyenURL(method)
+	url := a.adyenURL(service, method) + "/" + service + "/"
```
Comment (zhutik) [inline adyen.go]: Not sure if `service` should be added here. You already passing `service` to `adyenURL`  and it should return a URL like this `https://pal-test.adyen.com/pal/servlet/Recurring/v25/listRecurringDetails` or `https://pal-test.adyen.com/pal/servlet/Payment/v25/Capture`

File: adyen.go [hunk 3]
```diff
@@ -97,13 +103,13 @@ func (a *Adyen) SetCurrency(currency string) {
 }
 
 // execute request on Adyen side, transforms "requestEntity" into JSON representation
-func (a *Adyen) execute(method string, requestEntity interface{}) (*Response, error) {
+func (a *Adyen) execute(method string, service string, requestEntity interface{}) (*Response, error) {
```
Comment (zhutik) [inline adyen.go]: Maybe for consistency, we should change an order of params? now it's `method, service` but adyenURL accept `service, method`. I think `service, method` make more sense, cause we are moving from generic to the specific method that needs to be called.

Comment (lbailly) [reply adyen.go]: No problem, I'll push an update

File: adyen.go (hunk 2 above)
Comment (lbailly) [reply adyen.go]: Good catch I should have review a bit more the code ;-)

File: adyen.go [hunk 4]
```diff
@@ -71,8 +77,8 @@ func (a *Adyen) ClientURL() string {
 }
 
 // adyenURL returns Adyen backend URL
-func (a *Adyen) adyenURL(requestType string) string {
-	return a.Credentials.Env.BaseURL(APIVersion) + "/" + requestType
+func (a *Adyen) adyenURL(service string, requestType string) string {
+	return a.Credentials.Env.BaseURL(service, APIVersion) + "/" + requestType + "/" + service + "/"
```
Comment (zhutik) [inline adyen.go:81]: Seems like the issue with double `service` parameter doesn't solve. Can you please check?

**Comment by zhutik** [comment]
Merging, will fix `adyenURL` in separate commit

**Pull request merged**